app/src/main/java/com/kms/cura/controller/ErrorController.java

```diff
@@ -22,7 +23,10 @@ public void onClick(DialogInterface dialog, int which) {
                 dialog.dismiss();
             }
         });
-        dialog.show();
+        if(!((Activity) mContext).isFinishing())
```

Comment:
Also, should extract `!((Activity) mContext).isFinishing()` into a method to re-use below